### PR TITLE
[clang-tidy] fix performance-* warnings

### PR DESCRIPTION
**Pull request by Rechi**
## Description
fix performance-* clang-tidy warnings

## Types of change
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by howie-f — APPROVED** [review]
makes perfect sense

**Review by garbear — APPROVED** [review]
Game changes look good

File: xbmc/FileItem.cpp [hunk 1]
```diff
@@ -3334,7 +3334,7 @@ std::string CFileItem::GetThumbHideIfUnwatched(const CFileItem* item) const
                                     CSettings::VIDEOLIBRARY_THUMB_SHOW_UNWATCHED_EPISODE) &&
       item->HasArt("thumb"))
   {
-    const std::string fanArt = item->GetArt("fanart");
+    std::string fanArt = item->GetArt("fanart");
```
Comment (ksooo) [inline xbmc/FileItem.cpp:3337]: why this change?

File: xbmc/pvr/epg/EpgDatabase.cpp [hunk 2]
```diff
@@ -387,7 +387,7 @@ std::shared_ptr<CPVREpgInfoTag> CPVREpgDatabase::CreateEpgTag(
 {
   if (!pDS->eof())
   {
-    const std::shared_ptr<CPVREpgInfoTag> newTag(
+    std::shared_ptr<CPVREpgInfoTag> newTag(
```
Comment (ksooo) [inline xbmc/pvr/epg/EpgDatabase.cpp:390]: why this change?

File: xbmc/pvr/epg/EpgDatabase.cpp [hunk 3]
```diff
@@ -1234,7 +1234,7 @@ std::shared_ptr<CPVREpgSearchFilter> CPVREpgDatabase::CreateEpgSearchFilter(
 {
   if (!pDS->eof())
   {
-    const auto newSearch = std::make_shared<CPVREpgSearchFilter>(bRadio);
+    auto newSearch = std::make_shared<CPVREpgSearchFilter>(bRadio);
```
Comment (ksooo) [inline xbmc/pvr/epg/EpgDatabase.cpp:1237]: why?

File: xbmc/pvr/epg/EpgDatabase.cpp [hunk 4]
```diff
@@ -1311,7 +1311,7 @@ std::shared_ptr<CPVREpgSearchFilter> CPVREpgDatabase::GetSavedSearchById(bool bR
   {
     try
     {
-      const std::shared_ptr<CPVREpgSearchFilter> filter = CreateEpgSearchFilter(bRadio, m_pDS);
+      std::shared_ptr<CPVREpgSearchFilter> filter = CreateEpgSearchFilter(bRadio, m_pDS);
```
Comment (ksooo) [inline xbmc/pvr/epg/EpgDatabase.cpp:1314]: why?

File: xbmc/FileItem.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/FileItem.cpp:3337]: nvm, found it.

Comment (Rechi) [reply xbmc/FileItem.cpp:3337]: If this variable is const, it cannot be moved at the return statement.

see https://clang.llvm.org/extra/clang-tidy/checks/performance-no-automatic-move.html & https://github.com/xbmc/xbmc/pull/19882#discussion_r652444778
